treeherder/model/derived/jobs.py

```diff
@@ -1250,6 +1257,81 @@ def store_job_data(self, data, raise_errors=False):
                 placeholders=coalesced_job_guid_placeholders,
                 executemany=True)
 
+    def store_possible_job_data(self, data, raise_errors=False):
+        """
```

Comment:
I think you need a strategy to modify the possible jobs when the list changes upstream